x/orderbook/msg.go

```diff
@@ -34,60 +36,58 @@ func (CancelOrderMsg) Path() string {
 }
 
 func (m CreateOrderBookMsg) Validate() error {
-	if err := m.Metadata.Validate(); err != nil {
-		return errors.Wrap(err, "metadata")
-	}
-	if err := validID(m.MarketID); err != nil {
-		return errors.Wrap(err, "market id")
-	}
+	var errs error
+
+	errs = errors.AppendField(errs, "metadata", m.Metadata.Validate())
+	errs = errors.AppendField(errs, "market id", validID(m.MarketID))
+
 	if !coin.IsCC(m.AskTicker) {
-		return errors.Wrapf(errors.ErrCurrency, "Invalid Ask Ticker: %s", m.AskTicker)
+		errs = errors.Append(errs,
+			errors.Field("AskTicker", errors.ErrCurrency, fmt.Sprintf("Invalid ask ticker: %s", m.AskTicker)))
 	}
 	if !coin.IsCC(m.BidTicker) {
-		return errors.Wrapf(errors.ErrCurrency, "Invalid Bid Ticker: %s", m.BidTicker)
+		errs = errors.Append(errs,
+			errors.Field("BidTicker", errors.ErrCurrency, "Invalid bid ticker"))
 	}
 	if m.BidTicker <= m.AskTicker {
-		return errors.Wrapf(errors.ErrCurrency, "ask (%s) must be before bid (%s)", m.AskTicker, m.BidTicker)
+		errs = errors.Append(errs,
+			errors.Field("BidTicker", errors.ErrCurrency, fmt.Sprintf("ask (%s) must be before bid (%s)", m.AskTicker, m.BidTicker)))
```

Comment:
[`errors.Field`](https://godoc.org/github.com/iov-one/weave/errors#Field) can format description for you, no need to use `fmt.Sprintf`.
Similar in other cases.